### PR TITLE
Add an api to retrigger all the builds for a commit.

### PR DESCRIPTION
**Pull request by godofredoc**
This API is required to trigger builds for rollers where is not possible
to upload a new commit and for some reason the checks API failed to
update the status.

Bug:
  https://github.com/flutter/flutter/issues/61904

### CODE REVIEW

File: app_dart/bin/server.dart [hunk 1]
```diff
@@ -67,17 +73,31 @@ Future<void> main() async {
...
+          RefreshChromebotStatus(config, authProvider),
       '/api/refresh-github-commits': RefreshGithubCommits(config, authProvider),
       '/api/refresh-cirrus-status': RefreshCirrusStatus(config, authProvider),
       '/api/reserve-task': ReserveTask(config, authProvider),
-      '/api/reset-devicelab-task': ResetDevicelabTask(config, authProvider),
+      '/api/reset-devicelab-task': ResetDevicelabTask(
+        config,
+        authProvider,
+      ),
+      '/api/reset-try-task': ResetTryTask(
```
Comment (keyonghan) [inline app_dart/bin/server.dart:91]: Could this be used to re-run prod task as well? If yes, then `reset-luci-task`?

Comment (godofredoc) [reply app_dart/bin/server.dart:91]: They actually require different code paths and different parameters.

File: app_dart/test/request_handlers/reset_try_task_test.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/test/request_handlers/reset_try_task_test.dart]: done

File: app_dart/lib/src/request_handlers/reset_try_task.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  @override
+  Future<Body> get() async {
+    final String owner = request.uri.queryParameters['owner'] ?? 'flutter';
+    final String repo = request.uri.queryParameters['repo'] ?? '';
+    final String pr = request.uri.queryParameters['pr'] ?? '';
+    final String commitSha = request.uri.queryParameters['commitSha'] ?? '';
+    if (<String>[repo, pr, commitSha]
+        .any((String element) => element.isEmpty)) {
+      throw const BadRequestException('Any of repo, pr or commitSha is empty');
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/lib/src/request_handlers/reset_try_task.dart]: done

File: app_dart/lib/src/request_handlers/reset_try_task.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+
+@immutable
+class ResetTryTask extends ApiRequestHandler<Body> {
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/lib/src/request_handlers/reset_try_task.dart:21]: done

File: app_dart/lib/src/request_handlers/reset_try_task.dart [hunk 5]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (godofredoc) [reply app_dart/lib/src/request_handlers/reset_try_task.dart]: done

**Comment by godofredoc** [comment]
> dartfmt has been updated to use `--line-length=120` for both app_flutter and app_dart dirs.

Thanks for the info, VS code keeps reformatting to 80 chars

**Pull request merged**